### PR TITLE
Merge branch 'develop' of https://github.com/UniversityOfSaskatchewanCMPT371/term-project-fall2019-team-2 into ank266-dev

### PR DESCRIPTION
**Pull request by ank266**
isValid now works before sortData and sortData does not work when isValid says that data is not valid

### CODE REVIEW

File: src/__tests__/ParserComponent.test.tsx [hunk 1]
```diff
@@ -352,18 +352,26 @@ describe('<ParserComponent /> Unit Tests', () => {
...
-      expect(t1[0].drawType).toBe(enumDrawType.any);
-      // test date
-      // expect(t1[1].drawType).toBe(enumDrawType.any);
+      if (t1 !== undefined) {
+        // test string
+        expect(t1[2].drawType).toBe(enumDrawType.occurrence);
+        // test number
+        expect(t1[0].drawType).toBe(enumDrawType.any);
+        // test date
+        // expect(t1[1].drawType).toBe(enumDrawType.any);
```
Comment (eileenvanheerde) [inline src/__tests__/ParserComponent.test.tsx]: Why is this commented out?

File: src/__tests__/ParserComponent.test.tsx [hunk 2]
```diff
@@ -352,18 +352,26 @@ describe('<ParserComponent /> Unit Tests', () => {
...
       const t1 = pc.inferTypes(data);
+      if (t1 !== undefined) {
       // test string
-      expect(t1[2].drawType).toBe(enumDrawType.occurrence);
+        expect(t1[2].drawType).toBe(enumDrawType.occurrence);
+      }
+    });
+    it('give empty data', () => {
+      const data1 = new Array(0);
+      expect(data1).toEqual([]);
```
Comment (eileenvanheerde) [inline src/__tests__/ParserComponent.test.tsx]: I don't understand this test... You create an empty array an then check to see if it's empty. That doesn't actually test if, whatever function you want to test here, actually handles being passed in empty data

File: src/components/ParserComponent.tsx [hunk 3]
```diff
@@ -132,6 +132,7 @@ export default class ParserComponent extends React.Component<ParserInterface,
                 data,
               };
             });
+            // console.log(this.state.data);
```
Comment (eileenvanheerde) [inline src/components/ParserComponent.tsx]: get rid of these if you're not using them anymore

File: src/components/ParserComponent.tsx [hunk 4]
```diff
@@ -254,8 +260,10 @@ export default class ParserComponent extends React.Component<ParserInterface,
         };
       });
 
-      if (this.props.fileType === FileType.csv) {
+      const temp: File = fileEvent.target.files[0];
+      if (this.props.fileType === FileType.csv && this.isValid(temp)) {
```
Comment (eileenvanheerde) [inline src/components/ParserComponent.tsx]: Thank you for fixing this

File: src/components/ParserComponent.tsx [hunk 5]
```diff
@@ -254,8 +260,10 @@ export default class ParserComponent extends React.Component<ParserInterface,
...
         await this.parseCsv(fileEvent);
+        this.sortData(this.state.data);
```
Comment (eileenvanheerde) [inline src/components/ParserComponent.tsx]: did you remove the console.log() in parseCsv() that calls sortData()?

File: src/__tests__/ParserComponent.test.tsx (hunk 2 above)
Comment (ank266) [reply src/__tests__/ParserComponent.test.tsx]: this part is outdated, there should be new lines there

File: src/__tests__/ParserComponent.test.tsx (hunk 1 above)
Comment (ank266) [reply src/__tests__/ParserComponent.test.tsx]: removed the comment and it passed test so line 361 is not a comment now

File: src/components/ParserComponent.tsx (hunk 3 above)
Comment (ank266) [reply src/components/ParserComponent.tsx]: all lines that are code behind comments now deleted

File: src/components/ParserComponent.tsx (hunk 5 above)
Comment (ank266) [reply src/components/ParserComponent.tsx]: ya i removed it

**Pull request closed without merging**